### PR TITLE
Basic auth

### PR DESCRIPTION
**Pull request by donbobka**
Hi @randallagordon. Some features for your PR
1. Fixed tests. Moved `auth_basic` statements from `location` to `server` in nginx config 
2. If there is environment variable `BASIC_AUTH_USERNAME` then automaticaly enable basic_auth option. Added spec for that case.

### CODE REVIEW

**Comment by donbobka** [comment]
CircleCI build: https://circleci.com/gh/donbobka/heroku-buildpack-static/4

**Comment by donbobka** [comment]
Should add specs with valid username and password

**Comment by randallagordon** [comment]
Awesome, thanks so much @donbobka! I'll take a proper look over this later today and get it merged.

File: spec/simple_spec.rb [hunk 1]
```diff
@@ -155,10 +155,24 @@
...
-        app.run do
-          response = app.get("/foo.html")
-          expect(response.code).to eq("401")
-        end
+        response = app.get("/index.html")
+        expect(response.code).to eq("401")
+      end
+    end
+
+    context "static.json with static.json" do
```
Comment (randallagordon) [inline spec/simple_spec.rb]: Should this be `"static.json with basic_auth key"`?

**Pull request merged**